### PR TITLE
Make fetching PR Reviews paginated

### PR DESCRIPTION
**Pull request by lock14**
* Added pagination logic for review data. This ensures we don't miss any potential reviews for PRs with over 100 reviews.

### CODE REVIEW

File: pkg/review/commit_review_status_test.go [hunk 1]
```diff
@@ -90,8 +96,9 @@ func TestGetPullRequests(t *testing.T) {
            "variables": {
              "commitSha": "kof6p96lr6qvdu81qw49fhmoxrod9qmc2qak51nh",
              "githubOrg": "test-org",
-             "pageCursor": "",
-             "repository":"test-repo"
+             "pullRequestCursor": "",
+             "repository":"test-repo",
+			 "reviewCursor": null
```
Comment (verbanicm) [inline pkg/review/commit_review_status_test.go:101]: nit: formatting, indentation looks off here

File: pkg/review/commit_review_status.go [hunk 2]
```diff
@@ -147,9 +147,30 @@ type breakGlassIssue struct {
...
+type CommitGraphQlQuery struct {
+	Repository struct {
+		DefaultBranchRef struct {
+			Name githubv4.String
+		}
+		Object struct {
+			Commit struct {
+				AssociatedPullRequest struct {
+					Nodes      []*PullRequest
+					PageInfo   PageInfo
```
Comment (sethvargo) [inline pkg/review/commit_review_status.go:161]:
```suggestion
					PageInfo   *PageInfo
```

File: pkg/review/commit_review_status.go [hunk 3]
```diff
@@ -158,15 +179,29 @@ type PullRequest struct {
 	DatabaseID  githubv4.Int
 	Number      githubv4.Int
 	Reviews     struct {
-		Nodes []Review
-	} `graphql:"reviews(first: 100)"`
+		Nodes    []*Review
+		PageInfo PageInfo
```
Comment (sethvargo) [inline pkg/review/commit_review_status.go:183]:
```suggestion
		PageInfo *PageInfo
```

File: pkg/review/commit_review_status.go [hunk 4]
```diff
@@ -485,49 +520,50 @@ func formatGoogleSQL(qualifiedTableName *bigqueryio.QualifiedTableName) string {
...
-			"commitSha":  githubv4.GitObjectID(commitSha),
-			"pageCursor": cursor,
+		if err := client.Query(ctx, &query, map[string]interface{}{
+			"githubOrg":         githubv4.String(githubOrg),
+			"repository":        githubv4.String(repository),
+			"commitSha":         githubv4.GitObjectID(commitSha),
+			"pullRequestCursor": pullRequestCursor,
+			// The initial reviewCursor must be nil and not the empty string "",
+			// unlike the pullRequestCursor.
+			"reviewCursor": (*githubv4.String)(nil),
```
Comment (sethvargo) [inline pkg/review/commit_review_status.go:534]: Can you just omit this field entirely? That should be the same as setting it to a typed nil.

File: pkg/review/commit_review_status.go [hunk 5]
```diff
@@ -485,49 +520,50 @@ func formatGoogleSQL(qualifiedTableName *bigqueryio.QualifiedTableName) string {
...
-							StartCursor     githubv4.String
-						}
-						TotalCount githubv4.Int
-					} `graphql:"associatedPullRequests(first: 100, after: $pageCursor)"`
-				} `graphql:"... on Commit"`
-			} `graphql:"object(oid: $commitSha)"`
-		} `graphql:"repository(owner: $githubOrg, name: $repository)"`
-	}
-
+	query := CommitGraphQlQuery{}
```
Comment (sethvargo) [inline pkg/review/commit_review_status.go:523]:
```suggestion
	var query CommitGraphQlQuery
```

File: pkg/review/commit_review_status.go [hunk 6]
```diff
@@ -485,49 +520,50 @@ func formatGoogleSQL(qualifiedTableName *bigqueryio.QualifiedTableName) string {
...
 
-		for _, pr := range query.Repository.Object.Commit.AssociatedPullRequest.Nodes {
+		for i := 0; i < len(query.Repository.Object.Commit.AssociatedPullRequest.Nodes); i++ {
+			pr := query.Repository.Object.Commit.AssociatedPullRequest.Nodes[i]
 			if pr.BaseRefName == query.Repository.DefaultBranchRef.Name {
+				// We need to account for when reviewNodes span multiple pages.
+				for pr.Reviews.PageInfo.HasNextPage {
+					// Make a new query object so that our existing query's
+					// state is not obliterated.
+					reviewQuery := CommitGraphQlQuery{}
```
Comment (sethvargo) [inline pkg/review/commit_review_status.go:546]:
```suggestion
					var reviewQuery CommitGraphQlQuery
```

File: pkg/review/commit_review_status.go (hunk 4 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: We cannot due to having to annotate the struct with graphql call:
https://github.com/abcxyz/github-metrics-aggregator/blob/7cfa0926f3362c29837a4f0ea7762cba00f5e3e7/pkg/review/commit_review_status.go#L184

If we remove this we get the following error:
```
Variable $reviewCursor is used by anonymous query but not declared
```

File: pkg/review/commit_review_status.go (hunk 5 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: done

File: pkg/review/commit_review_status.go (hunk 6 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: done

File: pkg/review/commit_review_status.go (hunk 3 above)
Comment (lock14) [reply pkg/review/commit_review_status.go:183]: done.

File: pkg/review/commit_review_status.go (hunk 2 above)
Comment (lock14) [reply pkg/review/commit_review_status.go:161]: done

File: pkg/review/commit_review_status_test.go (hunk 1 above)
Comment (lock14) [reply pkg/review/commit_review_status_test.go]: fixed

**Review by verbanicm — APPROVED** [review]

**Pull request merged**